Update Wasm page for current status after finalized WasmGC encodings (#9711)
flutter/website: src/platform-integration/web/wasm.md

---
title: Support for WebAssembly (Wasm)

> description: >-

Reviewer: Nice!

  Current status of Flutter's experimental support for WebAssembly (Wasm).
short-title: Wasm
last-update: October 31, 2023
---

The Flutter and Dart teams are excited to add
[WebAssembly](https://webassembly.org/) as a compilation target when building
applications for the web.

{{site.alert.warning}}
  Development of WebAssembly support for Dart and Flutter remains ongoing.
  
  You can preview Wasm and WebAssembly garbage collection (WasmGC)
  on the [`master` channel][].

>   As Flutter provides these features as previews,
>   ongoing development might result in frequent changes.
>   Revisit this page for the latest updates.

Reviewer 2: **suggestion:** This reads a little awkward.
```suggestion
  Flutter is previewing these features.
  Ongoing development might result in frequent changes.
  Revisit this page for the latest updates.
```

  
  **_Last updated {{page.last-update}}_**
{{site.alert.end}}

[`master` channel]: https://github.com/flutter/flutter/wiki/flutter-build-release-channels#master

### Background

> To compile Dart and Flutter to Wasm, you need

Reviewer: whitespace churn makes it hard to review! 😄

Author: Ah sorry, I updated to closer match our standards. I'll split it into different commits next time!

The main change in the PR is mentioning Chrome 119 and Firefox 120 without flags, and being more explicit about updating to the latest `main`.

a browser that supports [WasmGC][].
The Wasm standard plans to add WasmGC to help garbage-collected languages
like Dart execute code in an efficient manner.

[Chromium and V8][] released stable support for WasmGC in Chromium 119,
and the Firefox team plans to release stable support in Firefox 120.
To see the current status of WasmGC and other proposals,
check out the [WebAssembly roadmap][].

[WasmGC]: https://github.com/WebAssembly/gc/tree/main/proposals/gc
[Chromium and V8]: https://chromestatus.com/feature/6062715726462976
[WebAssembly roadmap]: https://webassembly.org/roadmap/

### Try it out

#### Switch to the Flutter `master` channel and upgrade

Wasm compilation is only available on the latest builds of the `master` channel.

To learn more about Flutter build release channels and how to switch to
the `master` channel, check out the
[Flutter wiki](https://github.com/flutter/flutter/wiki/Flutter-build-release-channels).

To then ensure you are running the latest version,
run `flutter upgrade`.

> To verify that you correctly set your environment to the `master` channel,

Reviewer 2: This could be written in a more concise way.

```suggestion
To verify if your Flutter install uses the `master` channel,
```

Verify and correctly are redundant.

run `flutter build web --help`.

> At the bottom of the output, you should find experimental wasm options like:

Reviewer 2: **issue:** Wasm is always initial caps.

#### Load it in a browser

As of {{page.last-update}},

> there are two known browser types that should run

Reviewer 2: **issue:** This could be written with greater clarity. 

```suggestion
two browser families should be able to run
```

Author: Nice wording!

Flutter/Wasm content:

- Chromium-based browsers

>   - Version 119 or greater.

Reviewer 2: **issue:** Google uses `later` in this case.

```suggestion
  - Version 119 or later.
```

- Firefox

>   - Version 120 or greater.

Reviewer 2: **issue:** Google uses `later` in this case.

```suggestion
  - Version 120 or later.
```

>     This currently includes the beta, nightly, and developer

Reviewer 2: **nit:** You should remove `currently`. All of our docs should be considered current, so you don't need this modifier.

Author: Good point.

This will change very soon or could change at any time, so I tried a different wording. Let me know what you think.
